framework/include/kernels/ScalarKernelBase.h

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ResidualObject.h"
+#include "ScalarCoupleable.h"
+#include "MooseVariableScalar.h"
+
+// Forward declarations
+class ScalarKernelBase;
+
+template <>
+InputParameters validParams<ScalarKernelBase>();
+
+class ScalarKernelBase : public ResidualObject, public ScalarCoupleable
+{
+public:
+  static InputParameters validParams();
+
+  ScalarKernelBase(const InputParameters & parameters);
+
+  virtual void reinit() = 0;
+
+  /**
+   * The variable that this kernel operates on.
+   */
+  virtual const MooseVariableScalar & variable() const override { return _var; }
+
+  /**
+   * Use this to enable/disable the scalar kernel
+   * @return true if the scalar kernel is active
+   */
+  virtual bool isActive() { return true; }
+
+  /**
+   * Retrieves the old value of the variable that this ScalarKernelBase operates on.
+   *
+   * Store this as a _reference_ in the constructor.
+   */
+  const VariableValue & uOld() const;
```

Comment:
Do you know why we have this? I assume this was just copy-pasted from `ScalarKernel`. Just curious if you know because I don't haha